clang/lib/CodeGen/CGCall.cpp

```diff
@@ -825,6 +831,47 @@ void computeSPIRKernelABIInfo(CodeGenModule &CGM, CGFunctionInfo &FI);
 }
 } // namespace clang
 
+ABIArgInfo CodeGenTypes::convertABIArgInfo(const llvm::abi::ABIArgInfo &abiInfo,
+                                           QualType type) {
+  ABIArgInfo result;
+
+  if (abiInfo.isDirect()) {
+    llvm::Type *CoercedType = nullptr;
+    if (abiInfo.getCoerceToType()) {
+      CoercedType = ReverseMapper.convertType(abiInfo.getCoerceToType());
+    }
+    if (!CoercedType) {
+      CoercedType = ConvertType(type);
```

Comment:
Why do we need this case? In which case is CoercedType not set but we still need it?